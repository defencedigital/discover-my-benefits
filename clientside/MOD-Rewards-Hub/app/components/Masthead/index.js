import React from 'react';
import PropTypes from 'prop-types';

const MH = props => <div className="masthead__bg" style={{ backgroundImage: `url(${props.imgSrc})` }} />;

MH.propTypes = {
  imgSrc: PropTypes.string,
};

export default MH;
