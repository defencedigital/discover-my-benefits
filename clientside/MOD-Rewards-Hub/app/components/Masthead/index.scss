.masthead__bg {
  width: 100%;
  position: relative;
  background-size: cover;
  background-position: center top;
  overflow: hidden;
  height: auto;
  min-height: 200px;
}

@include media-breakpoint-up(md) {
  .masthead__bg {
    min-height: 500px;
  }
}

.masthead-wrap {
  position: relative;
}
