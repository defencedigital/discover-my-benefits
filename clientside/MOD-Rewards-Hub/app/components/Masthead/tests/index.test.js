import React from 'react';
import { fromJS } from 'immutable';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Masthead from '../index';

Enzyme.configure({ adapter: new Adapter() });

const mockedProps = fromJS({
  imgSrc: 'imgSrc.png',
});

describe('Masthead component', () => {
  it('matches the snapshot', () => {
    const html = shallow(<Masthead {...mockedProps.toJS()} />);
    expect(html).toMatchSnapshot();
  });
});
