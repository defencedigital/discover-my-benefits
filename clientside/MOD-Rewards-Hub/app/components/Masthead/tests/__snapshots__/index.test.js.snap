// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Masthead component matches the snapshot 1`] = `ShallowWrapper {}`;
